x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/utils/NativeMemoryCalculatorTests.java

```diff
@@ -49,6 +54,49 @@ public void testAllowedBytesForMLWhenAutoIsFalse() {
         }
     }
 
+    public void testConsistencyInAutoCalculation() {
+        for (Tuple<Long, Long> nodeAndJvmSize : Arrays.asList(
+            Tuple.tuple(1073741824L, 432013312L), // 1GB and true JVM size
+            Tuple.tuple(2147483648L, 536870912L), // 2GB ...
```

Comment:
Does this test fail for any of 0.9GB, 1.1GB, 1.9GB and 2.1GB?  If it does then we may have a problem that will hit us in production.

Reply:
@droberts195 please see lines: https://github.com/elastic/elasticsearch/pull/74691/files#diff-633121fb7250eeda7eb60908a4d89e0d0007a60c2e01fe452c6e962933208822R72

I add and subtract 10MB from every node value and make sure things still pass. 

Additionally, we never allow things to be below 1GB. When autoscaling, that is the smallest size we will ever ask for.

Reply:
@droberts195  I will additionally add a clause where the trueJvm isn't provided for either the native bytes size nor the inverse check. This way we use the dynamic jvm calculation.

Everything still checksout fine.

Reply:
> please see lines: https://github.com/elastic/elasticsearch/pull/74691/files#diff-633121fb7250eeda7eb60908a4d89e0d0007a60c2e01fe452c6e962933208822R72

Sorry, I didn't look closely enough

> Additionally, we never allow things to be below 1GB. When autoscaling, that is the smallest size we will ever ask for.

The situation I am thinking of is where Docker reports a container as 1GB but the programs running inside it think the machine they're running on has 0.9999GB.  But it sounds like this is covered.

This PR LGRM to merge then 🚀